uktrade/data-hub-api
# Modify `EYBLead` model fields, serializer, and views for MVP

### Description of change

This PR makes changes to the `EYBLead` model, including adding placeholder fields that we are expecting to receive from EYB and creating relationships to other models (Sectors, Company, etc). The factory has been updated to reflect these changes.

Additionally, various modifications have been made to the serialiser and views, including custom functionality to translate string names of the related fields to actual model instances, and vice versa. For example, we expect to receive the the field `sector='Mining'`; the serialiser will then validate this points to a sector instance, get the instance, and link accordingly.

> [!IMPORTANT]
> The changes made in this PR were broken down into 2 sub-PRs, which have already been approved.
> - #5618
> - #5619 

### Testing Instructions

You should be able to POST an EYB lead (with Hawk credentials) to `v4/investment-lead/eyb` and it create a linked instance in the database. To get dummy data, you can use the factory:

```python
from datahub.investment_lead.test.factories import EYBLeadFactory
from datahub.investment_lead.serializers import CreateEYBLeadSerializer
from rest_framework.renderers import JSONRenderer

eyb_lead = EYBLeadFactory()
serializer = CreateEYBLeadSerializer(eyb_lead)
json = JSONRenderer().render(serializer.data)
json
```

### Checklist


  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>


### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.
